internal/core/src/segcore/segment_c.cpp

```diff
@@ -150,6 +150,43 @@ Insert(CSegmentInterface c_segment,
     }
 }
 
+CStatus
+InsertColumnData(CSegmentInterface c_segment,
+                 int64_t reserved_offset,
+                 int64_t size,
+                 const int64_t* row_ids,
+                 const uint64_t* timestamps,
+                 void* raw_data,
+                 int64_t count) {
+    try {
+        auto segment = (milvus::segcore::SegmentGrowing*)c_segment;
+        milvus::segcore::ColumnBasedRawData dataChunk{};
+
+        auto& schema = segment->get_schema();
+        auto sizeof_infos = schema.get_sizeof_infos();
+        dataChunk.columns_ = std::vector<milvus::aligned_vector<uint8_t>>(schema.size());
+        // reverse space for each field
+        for (int fid = 0; fid < schema.size(); ++fid) {
+            auto len = sizeof_infos[fid];
+            dataChunk.columns_[fid].resize(len * size);
+        }
```

Comment:
How to insert a column of variable-length strings?

Reply:
The plan is to insert column data via an interface similar to `SegmentSealedImpl::LoadFieldData`, 
* on the Go side, pass the marshaled byte slice for each field when calling cgo
* on the C side, unmarshal to extract each field 
which will be implemented the in near future.